docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -378,6 +381,30 @@ The performance overhead is dependent on the number of function’s instrumented
 
 ---
 
+### Implement control flow integrity checks
+
+| Compiler Flag                                                                                            | Supported since  | Description                                                  |
+|:-------------------------------------------------------------------------------------------------------- |:-------------:|:------------------------------------------------------------ |
+| <span id="-fcf-protection=full">`-fcf-protection=full`</span><br/>                                   |     GCC      <br/>Clang            | Enable branch protection to counter Return Oriented Programming (ROP) and Jump Oriented Programming (JOP)attacks on Intel |
+| <span id="-mbranch-protection-standard">`-mbranch-protection=standard`</span>                        |     GCC      <br/>Clang            | Enable branch protection to counter Return Oriented Programming (ROP) and Jump Oriented Programming (JOP) attacks on AArch64 |
+
+#### Synopsis
+
+Return-oriented programming (ROP) uses an initial subversion (such as a buffer overflow) to perform an indirect jump that executes an arbitrary sequence of instructions. A countermeasure is to ensure that jump addresses and return addresses are correct. This is not a complete solution, but it makes attacks harder to perform.
+
+#### Performance implications
+
+There are performance implications but they are typically mild
+due to hardware assistance.
```

Comment:
new line?

Reply:
Newlines will be fixed. I'll tweak the paragraph while I'm at it.